File: output/llama3.1-8b/avatar/Python/Java/codeforces_146_A.java
import java.util.Scanner;

public class codeforces_146_A {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        int n = Integer.parseInt(scanner.nextLine());
        String input = scanner.nextLine();
        int[] number = new int[input.length()];
        for (int i = 0; i < input.length(); i++) {
            number[i] = Character.getNumericValue(input.charAt(i));
        }
        int h1 = 0;
        int h2 = 0;
        boolean found = false;
        for (int i = 0; i < n; i++) {
            if (number[i] != 4 && number[i] != 7) {
                System.out.println("NO");
                found = true;
                break;
            }
            if (i < n / 2) {
                h1 += number[i];
            } else {
                h2 += number[i];
            }
        }
        if (!found) {
            if (h1 == h2) {
                System.out.println("YES");
            } else {
                System.out.println("NO");
            }
        }
    }
}

File: output/llama3.1-8b/avatar/Java/Python/codeforces_146_A.py
import math

def main():
    n = int(input())
    str = input()
    l = list(str)
    
    t = 0
    for i in range(n):
        if l[i] not in ['4', '7']:
            t = 1
    
    if t == 1:
        print("NO")
    else:
        x = sum(int(c) for c in l[:n//2])
        y = sum(int(c) for c in l[n//2:][::-1])
        
        if x == y:
            print("YES")
        else:
            print("NO")

if __name__ == "__main__":
    main()